helper/helper_test.go

```diff
@@ -98,4 +295,234 @@ var _ = Describe("helper tests", func() {
+			}
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for resize and no single data file ", func() {
+			Expect(*singleDataFile).To(Equal(false))
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for reduce cluster and no single data file ", func() {
+			Expect(*singleDataFile).To(Equal(false))
+			*destSize, *origSize = *origSize, *destSize
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+	})
+	Describe("RestoreReader tests", func() {
+		It("waitForPlugin normal completion", func() {
+			test_cmd1 := testPluginCmd{hasProcess_: true}
+			test_reader := new(RestoreReader)
+			test_reader.pluginCmd = &test_cmd1
+
+			err := test_reader.waitForPlugin()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(test_cmd1.waitCount).To(Equal(1))
+
+			// Check that waitForPlugin do nothing when no cmd and/or no process
+			test_cmd2 := testPluginCmd{hasProcess_: false}
+			test_reader.pluginCmd = &test_cmd2
+			err = test_reader.waitForPlugin()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(test_cmd2.waitCount).To(Equal(0))
+		})
+		It("waitForPlugin error in Wait happened", func() {
+			msg := "Expected test error"
+			test_cmd1 := testPluginCmd{hasProcess_: true, error: &msg}
+			test_reader := new(RestoreReader)
+			test_reader.pluginCmd = &test_cmd1
+
+			err := test_reader.waitForPlugin()
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal(msg))
+		})
+	})
 })
+
+func writeTestTOC(tocFile string) {
+	// Write test TOC. We are not going to read data using it, so dataLength is a random number
+	dataLength := 100
+	customTOC := fmt.Sprintf(`dataentries:
+1:
+    startbyte: 0
+    endbyte: 18
+2:
+    startbyte: 18
+    endbyte: %[1]d
+3:
+    startbyte: %[1]d
+    endbyte: %d
+`, dataLength+18, dataLength+18+18)
+	fToc, err := os.Create(tocFile)
+	Expect(err).ShouldNot(HaveOccurred())
```

Comment:
if error occurred then file is not closed!

Reply:
if error has occurred, then file was not opened